Modules/SaltStackConfig/SaltStackConfig.psm1

```diff
@@ -289,3 +291,346 @@ Function Get-SscActivity {
+  <#
+    .NOTES
+    ===========================================================================
+      Created by:	Brian Wuchner
+      Date:		February 12, 2022
+      Blog:		www.enterpriseadmins.org
+      Twitter:		@bwuch
+    ===========================================================================
+    .SYNOPSIS
+      This wrapper function will set minion key state information for SaltStack Config.
+    .DESCRIPTION
+      This wrapper function will call Get-SscData minions.set_minion_key_state and update the states for specific minions.  
+    .EXAMPLE
+      PS C:\> Get-SscMinionKeyState |?{$_.name -eq 'server2022a'} | Set-SscMinionKeyState -state accept
+    .EXAMPLE
+      PS C:\> Set-SscMinionKeyState -master 'salt' -minion 'server2022a' -state reject -confirm:$false
+  #>
+  [cmdletbinding(SupportsShouldProcess)]
+  param(
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$master,
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$minion,
+    [Parameter(Mandatory=$true)][ValidateSet('accept','reject')][string]$state
+  )
+
+  begin {
+    $collection = @()
+  }
+
+  process {
+    if ($PSCmdlet.ShouldProcess("$master : $minion" , $state)) {
+      $collection += ,@($master, $minion)
+    }
+  }
+  
+  end {
+    $kwarg = @{}
+    $kwarg.Add('state', $state)
+    if ($state -eq 'reject') {$kwarg.Add('include_accepted', $true)}
+    if ($state -eq 'accept') {$kwarg.Add('include_rejected', $true)}
+    if ($state -eq 'accept' -OR $state -eq 'reject') {$kwarg.Add('include_denied',$true)}
+    $kwarg.Add('minions', @( $collection ) )
+
+    (Get-SscData minions set_minion_key_state $kwarg).task_ids
+  }
+}
+
+Function Remove-SscMinionKeyState {
+  <#
+    .NOTES
+    ===========================================================================
+      Created by:	Brian Wuchner
+      Date:		February 12, 2022
+      Blog:		www.enterpriseadmins.org
+      Twitter:		@bwuch
+    ===========================================================================
+    .SYNOPSIS
+      This wrapper function will delete a minion key for SaltStack Config.
+    .DESCRIPTION
+      This wrapper function will call Get-SscData minions.set_minion_key_state and remove the specified minion keys.  
+    .EXAMPLE
+      PS C:\> Get-SscMinionKeyState |?{$_.name -eq 'server2022a'} | Remove-SscMinionKeyState
+    .EXAMPLE
+      PS C:\> Remove-SscMinionKeyState -master 'salt' -minion 'server2022a' -confirm:$false
+  #>
+  [cmdletbinding(SupportsShouldProcess=$true,ConfirmImpact='High')]
+  param(
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$master,
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)][string]$minion
+  )
+
+  begin {
+    $collection = @()
+  }
+
+  process {
+    if ($PSCmdlet.ShouldProcess("$master : $minion" , 'delete')) {

```

Comment:
When the minion key is removed, the minion no longer shows up in the UI.  However, once the minion service is restarted on the remote machine it will appear again.

Reply:
Looking at the UI above I still think that Remove-SscMinionKey is more appropriate.

Reply:
I see, I'll get this fixed in the next commit.